Overhaul the `BASICS` part of the documentation (#50932)
LLNL/spack: lib/spack/docs/getting_started.rst

         $ . spack/share/spack/setup-env.fish

Now you're ready to use Spack!

-----------------------------
List packages you can install
-----------------------------

Once Spack is ready you can list all the packages it knows about with the following command:

.. code-block:: console

   $ spack list

If you want to get more information on a specific package, for instance ``hdf5``, you can use:

.. code-block:: console

   $ spack info hdf5

This command shows information about ``hdf5``, including a brief description, the versions of the package Spack knows about, and all the options you can activate when installing.

As you can see it's quite simple to gather basic information on packages, before you install them!

.. admonition:: Slowdown on the very first command
   :class: warning
   :collapsible:

>    The very first command run with Spack will take a while to finish, as Spack has to build a few caches to speed-up further command execution.

Reviewer: ```suggestion
   The very first command run with Spack will take a while to finish, as Spack has to build a few caches to speed up subsequent command execution.
```

>    This will be just a one-off slowdown though, and subsequent command execution is much faster.

Reviewer: ```suggestion
```

That's already clear, don't spend too many words on it

-----------------------------
Installing your first package
-----------------------------

To install most packages, Spack needs a compiler suite to be available.
To search your machine for available compilers, you can run:

.. code-block:: console

   $ spack compiler find

The command shows users if any compiler was found, and where its configuration is stored.
If the search was successful, you can now list known compilers, and get an output similar to the following:

.. code-block:: console

   $ spack compiler list
   ==> Available compilers
   -- gcc ubuntu20.04-x86_64 ---------------------------------------
   [e]  gcc@9.4.0  [e]  gcc@8.4.0  [e]  gcc@10.5.0

> If no compiler was found, you need either to:

Reviewer: ```suggestion
If no compilers were found, you need either to:
```

* Install further prerequisites, see :ref:`verify-spack-prerequisites`, and repeat the search above.

> * Register a buildcache that provides a compiler already available as a binary

Reviewer: That's not helpful advice for "getting started", since we do not provide this conveniently.

Author: We have the devtools buildcaches, which are available for both linux and macOS

   [+] /home/spack/.local/spack/opt/linux-icelake/zlib-ng-2.2.4-j5ddfaq7nyykn2bovorx73gykhjcl5nz
   ==> No binary for tcl-8.6.12-6vo5hxeqw5plzd6gvzm74wlfz5stnzcv found: installing from source
   ==> Installing tcl-8.6.12-6vo5hxeqw5plzd6gvzm74wlfz5stnzcv [7/7]
   ==> Fetching https://mirror.spack.io/_source-cache/archive/26/26c995dd0f167e48b11961d891ee555f680c175f7173ff8cb829f4ebcde4c1a6.tar.gz
       [100%]   10.35 MB @   48.5 MB/s
   ==> No patches needed for tcl
   ==> tcl: Executing phase: 'autoreconf'
   ==> tcl: Executing phase: 'configure'
   ==> tcl: Executing phase: 'build'
   ==> tcl: Executing phase: 'install'
   ==> tcl: Successfully installed tcl-8.6.12-6vo5hxeqw5plzd6gvzm74wlfz5stnzcv
     Stage: 0.46s.  Autoreconf: 0.00s.  Configure: 9.25s.  Build: 1m 8.71s.  Install: 3.32s.  Post-install: 0.68s.  Total: 1m 22.61s
   [+] /home/spack/.local/spack/opt/linux-icelake/tcl-8.6.12-6vo5hxeqw5plzd6gvzm74wlfz5stnzcv

Congratulations! You just installed your first package with Spack!

-----------------------------------
Use the software you just installed
-----------------------------------

> Once you installed ``tcl``, you can immediately use it starting the ``tcsh`` with its absolute path:

Reviewer: ```suggestion
Once you have installed ``tcl``, you can immediately use it by starting the ``tclsh`` with its absolute path:
```

.. code-block:: console

   $ /home/spack/.local/spack/opt/linux-icelake/tcl-8.6.12-6vo5hxeqw5plzd6gvzm74wlfz5stnzcv/bin/tclsh
   >% echo "Hello world!"
   Hello world!

> This works because of how Spack builds packages, setting the ``RPATH`` of their dependencies, but using such a long absolute path is not the most ergonomic way to refer to an executable.

Reviewer: ```suggestion
This works, but using such a long absolute path is not the most convenient way to run an executable.
```

> The simplest way to have ``tcsh`` directory added to your ``PATH`` is to:

Reviewer: ```suggestion
The simplest way to make ``tclsh`` available on the command line is:
```

.. code-block:: console

   $ spack load tcl

> Now the environment of the current shell has been modified, and you can use:

Reviewer: ```suggestion
The environment of the current shell has now been modified, and you can run:
```

.. code-block:: console

   $ tcsh

directly. To undo these modifications, you can:

.. code-block:: console

   $ spack unload tcl

.. admonition:: Environments and views
   :class: tip

   :ref:`Spack Environments <spack-environments-basic-usage>` are a better way to install and load a set of packages that are frequently used together.
   The discussion of this topic goes beyond this ``Getting Started`` guide, and we refer to :ref:`environments` for more information.

----------
Next steps
----------

> This section just helped you getting Spack installed, and running, quickly.

Reviewer: ```suggestion
This section helped you get Spack installed and running quickly.
```

> There are further resources in the documentation, that cover both basic and advanced topics more in details:

Reviewer: ```suggestion
There are further resources in the documentation that cover both basic and advanced topics in more detail:
```
